### PR TITLE
[METEOR-1205][FIX] Add position to getTargetPosition rather than always using current position

### PR DESCRIPTION
**Pull request by patrickcleeve2**
The current get target position either used a pre-defined metadata position (e.g. grid_1) or the current stage position. This meant that we could not calculate the transformed position for arbitrary stage positions (e.g. feature positions).

This PR adds a position arg to the getTargetPosition to allow for calculating these transforms.

### CODE REVIEW

File: src/odemis/acq/move.py [hunk 1]
```diff
@@ -104,11 +104,13 @@ def getCurrentPostureLabel(self, pos: Dict[str, float] = None) -> int:
         """
         pass
 
-    def cryoSwitchSamplePosition(self, target: int):
+    def cryoSwitchSamplePosition(self, target: int, pos: Dict[str, float] = None) -> model.CancellableFuture:
```
Comment (nandishjpatel) [inline src/odemis/acq/move.py:107]:
```suggestion
    def cryoSwitchSamplePosition(self, target: int, pos: Optional[Dict[str, float]] = None) -> model.CancellableFuture:
```

**Review by tmoerkerken — APPROVED** [review]

File: src/odemis/acq/move.py (hunk 1 above)
Comment (nandishjpatel) [reply src/odemis/acq/move.py]: Can you update the type for all `pos` args in this PR?

**Review by nandishjpatel — APPROVED** [review]

**Review by pieleric — COMMENTED** [review]
This looks fine to me. However, as discussed about the larger work on supporting multiple posture, I think eventually there should be a generic function that should take, in addition to the position, also the current posture.

**Comment by patrickcleeve2** [comment]
Closing, replaced by #2893

**Pull request closed without merging**